### DISCUSSION TITLE
Sharing a file system to the docker container in the HTCondor setup

### DISCUSSION THREAD

**Discussion by katilp** [GoogleCloudPlatform/cluster-toolkit#2550] [Q&A]
I would like to mount a shared filesystem other than the HTCondor scratch volume to the docker containers in the HTCondor setup.

I have followed the [instructions](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/docs/network_storage.md#mounting-via-use) and mounted successfully a shared filesystem `/shared` to the access and execute points.

I understand from the [HTCondor admin manual](https://htcondor.readthedocs.io/en/lts/admin-manual/setting-up-vm-docker-universes.html) that I should set up this mount with:

> An administrator of a machine can optionally make additional directories on the host machine readable and writable by a running container. To do this, the admin must first give an HTCondor name to each directory with the DOCKER_VOLUMES parameter. Then, each volume must be configured with the path on the host OS with the DOCKER_VOLUME_DIR_XXX parameter. Finally, the parameter DOCKER_MOUNT_VOLUMES tells HTCondor which of these directories to always mount onto containers running on this machine.

Presumably, in my case - if I want to mount the host `/shared` to the container `/shared`  - this would be

```
DOCKER_VOLUMES = SHARED
DOCKER_VOLUME_DIR_SHARED = /shared
DOCKER_MOUNT_VOLUMES = SHARED
```

Would this be doable somewhere during the HTCondor installation which I believe happens [here](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/community/modules/scripts/htcondor-install/files/install-htcondor.yaml)

Thanks for any advise!

**Comment by tpdownes** [comment]
I agree that we don't yet have a well integrated solution for adding extra HTCondor snippets into the final system configuration. I have tried a couple workarounds but have not landed upon a good solution. I will follow up next week with you.

Ideally, I think the configuration snippets would be suppliable not on HTCondor install module (which configures every image) but would be added to the access/central manager/execute point modules so that they can be configured separately. We do accept community pull requests if you are willing to do this.

**Comment by katilp** [comment]
Thanks!

In the same go, if I may, it would be great to have an option to store the docker images in that mounted volume.

In principle, that could go in [here](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/community/modules/scripts/htcondor-install/files/install-htcondor.yaml#L85) as something like

```
- name: Change Docker default storage
  copy:
    content: '{\n"data-root": "{{ mounted_dir }}"\n}'
    dest: /etc/docker/daemon.json
  when: mounted_dir is defined and some_condition is true
```

That's what I would need (with our 30 G image 😬), but I guess not everyone would want it so it would be good to have an option (`some_condition`) in an option. 

Does that make sense?
